src/views/forms/AccountManagementFormsNew/SlippageEditor.tsx

```diff
@@ -0,0 +1,153 @@
+
+import { FormInput } from '@/components/FormInput';
+import { Icon, IconName } from '@/components/Icon';
+import { InputType } from '@/components/Input';
+import { Output, OutputType } from '@/components/Output';
+import { ToggleGroup } from '@/components/ToggleGroup';
+import { WithConfirmationPopover } from '@/components/WithConfirmationPopover';
+
+enum EditorState {
+  Viewing = 'Viewing',
+  Selecting = 'Selecting',
+  Editing = 'Editing',
+}
+
+type ElementProps = {
+  slippage: number;
+  setIsEditing?: Dispatch<SetStateAction<boolean>>;
+  setSlippage: (slippage: number) => void;
+  disabled?: boolean;
+};
+
+export type SlippageEditorProps = ElementProps;
+
+export const SlippageEditor = ({
+  disabled,
+  slippage,
+  setIsEditing,
+  setSlippage,
+}: SlippageEditorProps) => {
+  const percentSlippage = slippage * 100;
+  const [slippageInputValue, setSlippageInputValue] = useState(percentSlippage.toString());
+  const [editorState, setEditorState] = useState(EditorState.Viewing);
+
+  const stringGetter = useStringGetter();
+
+  const toggleGroupRef = useRef<HTMLDivElement>(null);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    setIsEditing?.(editorState !== EditorState.Viewing);
+
+    if (editorState === EditorState.Selecting) {
+      toggleGroupRef?.current?.focus();
+    } else if (editorState === EditorState.Editing) {
+      inputRef?.current?.focus();
+    }
+  }, [editorState, setIsEditing]);
+
+  const onOpenChange = (isOpen: boolean) => {
+    if (!isOpen) {
+      setEditorState(EditorState.Viewing);
+    }
+  };
+
+  const onCancel = () => {
+    setEditorState(EditorState.Viewing);
+    setSlippageInputValue(percentSlippage.toString());
+  };
+
+  const onChangeSlippage = ({ value }: NumberFormatValues) => {
+    setSlippageInputValue(value);
+  };
+
+  const onConfirmSlippage = () => {
+    setEditorState(EditorState.Viewing);
+    setSlippage(Number(slippageInputValue) / 100);
+  };
+
+  const onSelectSlippage = (newSlippage: string) => {
+    if (newSlippage === 'custom') {
+      setEditorState(EditorState.Editing);
+    } else {
+      setSlippage(Number(newSlippage));
+
+      setEditorState(EditorState.Viewing);
+    }
+  };
+
+  if (disabled) {
+    return <Output type={OutputType.Percent} value={slippage} />;
+  }
+
+  return (
+    <WithConfirmationPopover
+      open={editorState !== EditorState.Viewing}
+      onOpenChange={onOpenChange}
+      align="end"
+      sideOffset={-22}
+      asChild
+      onCancel={onCancel}
+      onConfirm={editorState === EditorState.Editing ? onConfirmSlippage : undefined}
+      slotTrigger={
+        <button
+          onClick={() => setEditorState(EditorState.Selecting)}
+          tw="row gap-[0.5ch] underline"
+          type="button"
+          aria-label="edit slippage"
+        >
+          <Output type={OutputType.Percent} value={slippage} />
+          <Icon iconName={IconName.Pencil} />
+        </button>
```

Comment:
This will emit an Invalid dom nesting warning.

Button within button (slotTrigger is a child of a button)

Reply:
🔥 i had no idea. i'm updating the `WithConfirmationPopover` to just accept an `onClickTrigger` property then instead

Reply:
Instead of making it an on click can't we just update they styling on the Trigger?